wum-live-sync-automation/parallelTriggerJob/Jenkinsfile

```diff
@@ -0,0 +1,31 @@
+import groovy.json.JsonSlurper
+
+node {
+    
+    stage('runParallelProductJobs') {
+        
+        try {
+            def receivedJson = System.properties['UPDATE_LIST_JSON']
+            println "Received Json: " + receivedJson
+            def jsonSlurper = new JsonSlurper()
+            def resultJson = jsonSlurper.parseText(receivedJson)
+            jsonSlurper = null
+            
+            def branches = [:]
+            
+            for (int index = 0; index < resultJson.size(); index++) {
+                String jobName = resultJson[index].jobName;
+                String fileUrl = resultJson[index].updateFileUrl;
+                String fileName = resultJson[index].updateFileName;
+                branches["branch:"+ jobName] = {build job: jobName, parameters: [[$class: 'StringParameterValue', name: 'UPDATE_FILE_URL', value: fileUrl], [$class: 'StringParameterValue', name: 'UPDATE_FILE_NAME', value: fileName]]};
+                println "Added parallel ECS slave to run, jobName: " + jobName + ", updateFileUrl: " + fileUrl + ", updateFileName: " + fileName
+            }
+            resultJson = null
+            
+            parallel branches
```

Comment:
In the actual deployment, how many parallel branches are we going to have? We've seen that Jenkins UI becomes quite unresponsive when there are loads of parallel jobs.

Reply:
Good point, though I did not do a perf test, with ECS most of the load will be distributed to ECS slaves and as of @maheshika 's experience with the current Jenkins master node, this can be handled. I think, we will go with this normal implementation and in case, we see any need to ECS node usage controlling we will act on it.